### ISSUE TITLE
Do not install proprietary NVIDIA drivers on Optimus

### ISSUE THREAD

**Issue by Yoshi2889** [Antergos/Cnchi#596]
Do not offer to install NVIDIA proprietary drivers when Optimus is detected. This to avoid issues with unbootable systems.

We've received numerous reports on IRC that a reinstall was needed (or otherwise major fixes) if this option was selected, so I'd say this is quite urgent.

**Comment by karasu** [comment]
I do not know if this is doable. How do we now that an user has an Optimus setup?

**Comment by Yoshi2889** [comment]
xrandr --listproviders should list two providers. That's one way to detect.

lspci lists multiple devices - but this is not reliable as you can still have intel graphics and nvidia graphics both enabled in a desktop system.

**Comment by karasu** [comment]
If you have two nvidias installed (like a sli configuration or something like that), it would show up as two providers too?

**Comment by Yoshi2889** [comment]
Yes, however it would show up as the same driver. With PRIME (nouveau) you could watch for intel and nouveau drivers both existing.
I'm not sure how reliable this is, though.

This is MHWD's file: https://github.com/manjaro/mhwd-db/blob/master/pci/graphic_drivers/hybrid-intel-nvidia-bumblebee/MHWDCONFIG

**Comment by karasu** [comment]
That file will be very helpful, thanks!

**Comment by Mikaela** [comment]
Someone again encountered this at IRC and had tried to reinstall Antergos three times before coming to ask for help and hearing this :crying_cat_face:

**Comment by Yoshi2889** [comment]
Are we even being listened to? Calling for help in IRC seems to have no
effect at all, reporting critical issues affects nothing as they're just
left to rot...

Please fix ASAP as these things shouldn't happen or remove the
functionality if you can't think of a fix.

Op za 28 mei 2016 09:15 schreef Mikaela Suomalainen <
notifications@github.com>:

**Comment by karasu** [comment]
Yes, of course we check all issues.
I'll remove that option until we have the time to fix it.
Thanks.

**Comment by Yoshi2889** [comment]
*Bump* More cases of this are happening on IRC.

**Comment by karasu** [comment]
That's odd. If Intel and Nvidia are detected, nvidia feature should not be shown. Sadly, we have no hardware to test this on.

**Comment by Yoshi2889** [comment]
I'll fire the latest ISO up later today to test this

Op 23 jul. 2016 19:35 schreef "karasu" <notifications@github.com>:

**Comment by Yoshi2889** [comment]
Nope I didn't get it this time... Maybe something wrong in the setup of
said user

Op za 23 jul. 2016 21:13 schreef Rick Kerkhof <rick.2889@gmail.com>:

**Comment by Yoshi2889** [comment]
Okay it's back now, the proprietary graphics option shows in Cnchi 0.14.140.

**Comment by enlighter** [comment]
I have an Optimus Setup on my brand new laptop and I installed Antergos with Proprietary driver option. Now the installed systemfails to load lightdm and gets stuck there.

Here's my Xorg.0.log: http://pastebin.com/J5ULd1q4

Is there a way to fix the installation and how? I really donot want to reinstall as I have already setup my UEFI multiboot using Windows 10, antergos and ubuntu 16.04

**Comment by lots0logs** [comment]
@enlighter I cant see your log b/c its private. Could you post a [paste2](https://paste2.org) link instead? Also will probably need to see your systemd journal log.

**Comment by enlighter** [comment]
Sorry! Here's my re-pasted Xorg.0.log: https://paste2.org/M3HD4ts0

**Comment by Yoshi2889** [comment]
You need to swap out the nvidia-libgl package with mesa-libgl and install
bumblebee, it'll find the nvidia-libgl libraries by itself. The Intel Mesa
driver doesn't work with Nvidia-libgl.

Op di 27 sep. 2016 15:10 schreef Sushovan Mandal <notifications@github.com>:

**Comment by enlighter** [comment]
@Yoshi2889 Thanks for the suggestion, installing bumblebee worked, but only after removing the xorg config created by `nvidia-xconfig` command. Now it works with full Optimus.

**Comment by Yoshi2889** [comment]
Yeah even nowadays there should be no reason to generate a config file,
Xorg will automatically detect stuff.

Op wo 28 sep. 2016 13:38 schreef Sushovan Mandal <notifications@github.com>:

**Comment by Yoshi2889** [comment]
I just had a quick chat with the people in #nouveau:

> [20:44:00] <NanoSector> what is the most reliable way to detect if a system is Optimus or not?
> [20:44:42] <NanoSector> I thought of checking whether both Intel and NVIDIA are present, but that'd mean desktops with the integraded GPU enabled would also pass
> [21:14:00] <airlied> NanoSector: why do you care about optimus specificlly?
> [21:28:31] <NanoSector> airlied: asking for the Antergos installer, it needs a way to detect if a laptop has Optimus and then allow the installation of bumblebee
> [21:29:15] <airlied> you'd have to probe the ACPI tables
> [21:31:27] <karolherbst> NanoSector: bumblebeeisn (logged in 25 minutes 36 seconds ago)
> [21:31:38] <karolherbst> 'nt limited to optimus
> [21:31:45] <karolherbst> only bbswitch is mainly
> [21:33:09] <karolherbst> NanoSector: wouldn't it make sense to install bumblebee whenever there are two GPUs installed?
> [21:39:18] <NanoSector> karolherbst: you mean also on desktop systems?
> [21:43:54] <karolherbst> yeah
> [21:43:56] <karolherbst> why not?
> [21:44:08] <karolherbst> imagine somebody with intel main and nvidia dedicated, but without a display
> [21:44:30] <karolherbst> on nvidia
> [21:44:51] <karolherbst> NanoSector: or better: if the script detects multiple GPUs, promp for something or so
> [21:45:12] <karolherbst> let the user decide in this case
> [21:45:30] <NanoSector> we'll default to Nouveau but we show an option to install nvidia proprietary drivers
> [21:46:19] <NanoSector> I suppose we can modify it to also install bumblebee

**Comment by karasu** [comment]
I have to copy some of the nvidia-installer script code to cnchi... now I understand better how nouveau, nvidia and bumblebee works.... I hope I'll be able to fix this soon.

**Comment by enlighter** [comment]
Just wanted to let know, that the manual for Nvidia Optimus from https://antergos.com/wiki/hardware/graphics/bumblebee-for-nvidia-optimus/ worked perfectly for me. I have a core i3 and Geforce 930M laptop and the power saving patch was not required as the official archlinux repos contain bumblebee patch 15 (bumblebee 3.2.1-15)
